sdfObject/sdfthing-oma-device.sdf.json

```diff
@@ -0,0 +1,198 @@
+          },
+          "Factory_Reset": {
+            "label": "Factory Reset",
+            "description": "Perform factory reset of the LwM2M Device to make the LwM2M Device to go through initial deployment sequence where provisioning and bootstrap sequence is performed. This requires client ensuring post factory reset to have minimal information to allow it to carry out one of the bootstrap methods specified in section 5.2.3.  When this Resource is executed, \"De-register\" operation MAY be sent to the LwM2M Server(s) before factory reset of the LwM2M Device."
+          },
+          "Reset_Error_Code": {
+            "label": "Reset Error Code",
+            "description": "Delete all error code Resource Instances and create only one zero-value error code that implies no error, then re-evaluate all error conditions and update and create Resources Instances to capture all current error conditions."
+          }
+        },
+        "sdfThing": {
+          "Power_Source": {
+            "minItems": 0,
+            "sdfProperty": {
+              "Type": {
+                "label": "Type",
+                "description": "Type of the Available Power Source.0: DC power 1: Internal Battery 2: External Battery 3: Fuel Cell 4: Power over Ethernet 5: USB 6: AC (Mains) power 7: Solar The same Resource Instance ID MUST be used to associate a given Power Source (Resource ID:6) with its Present Voltage (Resource ID:7) and its Present Current (Resource ID:8)",
+                "writable": false,
+                "type": "integer",
+                "sdfChoice": {
+                    "DC power": { "const": 0 },
+                    "Internal Battery": { "const": 1 },
+                    "External Battery": { "const": 2 },
+                    "Fuel Cell": { "const": 3 },
+                    "Power over Ethernet": { "const": 4 },
+                    "USB": { "const": 5 },
+                    "AC (Mains) power": { "const": 6 },
+                    "Solar": { "const": 7 }
+                }
+              },
+              "Voltage": {
+                "label": "Voltage",
+                "description": "Present voltage for the Available Power Source.",
+                "writable": false,
+                "type": "integer",
+                "unit": "mV"
+              },
+              "Current": {
+                "label": "Current",
+                "description": "Present current for the Power Source.",
+                "writable": false,
+                "type": "integer",
+                "unit": "mA"
+              }
+            }
+          },
+          "ExtDevInfo": {
+            "label": "ExtDevInfo",
+            "description": "Reference to external \"Device\" object instance containing information. For example, such an external device can be a Host Device, which is a device into which the Device containing the LwM2M client is embedded. This Resource may be used to retrieve information about the Host Device.",
+            "sdfRef": ""
```

Comment:
As discussed, here `"sdfType" : "link"` is probably the best option, but probably needs some thinking.

Reply:
Updated according to my understanding of https://www.ietf.org/archive/id/draft-bormann-asdf-sdftype-link-00.html